src/Microsoft.OpenApi/Reader/OpenApiJsonReader.cs

```diff
@@ -90,15 +90,18 @@ public ReadResult Read(JsonNode jsonNode,
             // Validate the document
             if (settings.RuleSet != null && settings.RuleSet.Rules.Any())
             {
-                var openApiErrors = document.Validate(settings.RuleSet);
-                foreach (var item in openApiErrors.OfType<OpenApiValidatorError>())
-                {
-                    diagnostic.Errors.Add(item);
-                }
-                foreach (var item in openApiErrors.OfType<OpenApiValidatorWarning>())
+                var openApiErrors = document?.Validate(settings.RuleSet);
+                if(openApiErrors is not null)
                 {
```

Comment:
it'd probably be better to check whether the document is not null in the previous condition